src/App.jsx

```diff
@@ -14,70 +15,63 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const click = (good) => {
+    if (selectedGood === good) {
+      setGood(null);
+    } else {
+      setGood(good);
+    }
+  };
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {!selectedGood ? (
+          <p>No goods selected</p>
+        ) : (
+          <div className="title is-flex is-align-items-center">
+            <p>{`${selectedGood} is selected`}</p>
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
+              className="delete ml-3"
+              onClick={() => {
+                setGood(null);
+              }}
+            />
+          </div>
+        )}
+      </h1>
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              className={selectedGood
+                === good ? 'has-background-success-light' : ''}
+              onClick={() => click(good)}
```

Comment:
there should be key on top level element
use [classnames](https://www.npmjs.com/package/classnames) library here
button should listen to click, not whole table row